### PR TITLE
Feature/arc 1259 cp materiaalaanvragen

### PR DESCRIPTION
**Pull request by silkederudder-shd**
Type filter is not implemented yet because Christophe needs to check which filter should be used on this page. This will be covered in [ARC-1379](https://meemoo.atlassian.net/browse/ARC-1379).

https://user-images.githubusercontent.com/77958553/218744096-05e590b7-681e-4d67-8c36-3962746398c6.mov

https://user-images.githubusercontent.com/77958553/218744133-1e7706b6-bc96-462d-9031-1acd1b4cdbd9.mov

[ARC-1379]: https://meemoo.atlassian.net/browse/ARC-1379?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by tourlbr — APPROVED** [review]

**Review by WardVerc1 — APPROVED** [review]

**Comment by bertyhell** [comment]
OPTIONEEL: mss nog een border rond de dropdown menu, zodat er wat contrast is met de tabel erachter. Maar het staat niet in het design:
image

**Comment by bertyhell** [comment]
You should be able to get rid of the flicker to the loading state when changing the sort order, by telling react-query to keep the previous result until a new result is fetched.
```
export function useGetNavigationItems(): UseQueryResult<Record<string, NavigationInfo[]>> {
	return useQuery([QUERY_KEYS.getNavigationItems], () => NavigationService.getAll(), {
		keepPreviousData: true,
	});
}
```

https://user-images.githubusercontent.com/1710840/219339777-1d0487d5-f614-499e-b0dc-2e9da5eedd10.mp4

**Comment by bertyhell** [comment]
You should now be able to also use the object name for the name column since the backend PR is merged: https://github.com/viaacode/hetarchief-proxy/pull/292/files

**Comment by silkederudder-shd** [comment]
> You should now be able to also use the object name for the name column since the backend PR is merged: https://github.com/viaacode/hetarchief-proxy/pull/292/files

Only needed for default users, added this in: https://github.com/viaacode/hetarchief-client/pull/630 :D

**Comment by silkederudder-shd** [comment]
> OPTIONEEL: mss nog een border (zelfde als zoek veld: 1px solid #e6e6e6) rond de dropdown menu, zodat er wat contrast is met de tabel erachter. Maar het staat niet in het design: image

Border of zoals bij alle andere dropdowns, die shadow is ook een optie.

**Comment by bertyhell** [comment]
The columns are not according to the spec of the epic:


Elke materiaalaanvraag zal bestaan uit:

- naam: “<voornaam v. aanvrager> <achternaam v. aanvrager>”
- emailadres: “<email adres v. aanvrager>”
- datum aanvraag: “<datum v. aanvraag>” formaat: “dd mon. yyyy, hh:mm”
- type: “<type v. aanvraag zoals opgegeven bij het aanvragen>”

**Comment by silkederudder-shd** [comment]
> You should be able to get rid of the flicker to the loading state when changing the sort order, by telling react-query to keep the previous result until a new result is fetched.
> 
> ```ts
> export function useGetNavigationItems(): UseQueryResult<Record<string, NavigationInfo[]>> {
> 	return useQuery([QUERY_KEYS.getNavigationItems], () => NavigationService.getAll(), {
> 		keepPreviousData: true,
> 	});
> }
> ```

Already added the `keepPreviousData: true`. The flow of this page, is equal to the other simular pages such as users and user permissions. On those pages, the loading stat is also visible when sorting the data.

Should this be fixed for all pages? @bertyhell

File: src/modules/cp/const/material-requests.const.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+	page: withDefault(NumberParam, 1),
+};
+
+export const MATERIAL_REQUEST_TRANSLATIONS_BY_TYPE: Record<MaterialRequestType, string> = {
+	[MaterialRequestType.MORE_INFO]: tText('modules/cp/const/material-requests___type-more-info'),
+	[MaterialRequestType.REUSE]: tText('modules/cp/const/material-requests___type-reuse'),
+	[MaterialRequestType.VIEW]: tText('modules/cp/const/material-requests___type-view'),
+};
+
+export const MaterialRequestTableColumns = (): Column<MaterialRequest>[] => [
```
Comment (bertyhell) [inline src/modules/cp/const/material-requests.const.tsx:33]:
```suggestion
export const getMaterialRequestTableColumns = (): Column<MaterialRequest>[] => [
```

**Review by bertyhell — APPROVED** [review]

**Pull request merged**